### PR TITLE
Extract app service common codes from web app components

### PR DESCRIPTION
**Pull request by Flanker32**
Extract app service common codes from web app components

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionComboBox.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.microsoft.azure.toolkit.intellij.appservice.AppServiceComboBox;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.core.mvp.model.ResourceEx;
+import com.microsoft.azuretools.core.mvp.model.function.AzureFunctionMvpModel;
+import com.microsoft.azuretools.utils.WebAppUtils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FunctionComboBox extends AppServiceComboBox<FunctionComboBoxModel> {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionComboBox.java:36]: `FunctionAppComboBox` <-> `WebAppComboBox`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                final AppServiceComboBoxModel app = (AppServiceComboBoxModel) value;
+                if (index >= 0) {
+                    setText(getWebAppLabelText(app));
+                } else {
+                    setText(app.getAppName());
+                }
+            }
+        }
+
+        private String getWebAppLabelText(AppServiceComboBoxModel webApp) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:138]: WebApp -> AppService

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                               this.setItems(items);
+                               this.setLoading(false);
+                               this.resetDefaultValue(defaultValue, AppServiceComboBoxModel::isSameApp);
+                           }), (e) -> {
+                                   this.handleLoadingError(e);
+                               });
+    }
+
+    private void resetDefaultValue(@NotNull AppServiceComboBoxModel defaultValue, @NotNull BiPredicate<AppServiceComboBoxModel,
+            AppServiceComboBoxModel> comparator) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:74]: why not remove arg 'comparator' just like subscription

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:138]: and some other webapp in this class.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBoxModel.java [hunk 4]
```diff
@@ -52,4 +52,13 @@ public AppComboBoxModel(ResourceEx<T> resourceEx) {
         this.subscriptionId = resourceEx.getSubscriptionId();
         this.isNewCreateResource = false;
     }
+
+    public static boolean isSameApp(AppServiceComboBoxModel first, AppServiceComboBoxModel second) {
+        return StringUtils.equalsAnyIgnoreCase(first.resourceId, second.resourceId) ||
+                (StringUtils.equalsAnyIgnoreCase(first.appName, second.appName) &&
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBoxModel.java:58]: equalsIgnoreCase

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override
+    protected void createResource() {
+
+    }
+
+    @NotNull
+    @Override
+    protected List<? extends FunctionAppComboBoxModel> loadItems() throws Exception {
+        final List<ResourceEx<FunctionApp>> functions = AzureFunctionMvpModel.getInstance().listAllFunctions(false);
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java:50]: AzureFunctionMvpModel.getInstance().listAllFunctions have not perf issue?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        return functions.stream()
+                        .filter(resource -> WebAppUtils.isJavaWebApp(resource.getResource()))
+                        .sorted((a, b) -> a.getResource().name().compareToIgnoreCase(b.getResource().name()))
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java:53]: StringUtils.compareToIgnoreCase

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java (hunk 5 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/FunctionAppComboBox.java:51]: Yes, we have fixed it before

**Review by andxu — APPROVED** [review]

**Pull request merged**